ts/configuration/v1.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Exposes the Kubernetes extension's Configuration settings
+ */
+export interface ConfigurationV1 {
+    /**
+     * Gets the path for the kubeconfig currently in use.
+     * @returns a KubeconfigPath designating the environment the path relates to, and the path.
+     */
+    getKubeconfigPath(): ConfigurationV1.KubeconfigPath;
+}
+
+export namespace ConfigurationV1 {
+    /**
+     * Represents the path to the kubeconfig on the host machine.
+     */
+    export interface HostKubeconfigPath {
+        /**
+         * The environment the path relates to
+         */
+        readonly pathType: 'host';
+        /**
+         * The path to the kubeconfig
+         */
+        hostPath: string;
+    }
+
+    /**
+     * Represents the path to the kubeconfig on WSL.
+     */
+    export interface WslKubeconfigPath {
```

Comment:
Do we have a feeling for the TypeScript/JS capitalisation convention here?  I started out bring the `Wsl`/`Http` etc style from my .NET days, but of late I've felt that `WSL` and `HTTP` are more idiomatic in JS (hence e.g. API rather than Api).  I am probably not very consistent though!